n3fit/src/n3fit/layers/msr_normalization.py

```diff
@@ -51,30 +61,26 @@ def call(self, pdf_integrated, photon_integral):
 
         Parameters
         ----------
-        pdf_integrated: (Tensor(1,None,14))
+        pdf_integrated: (Tensor(1, 14))
             the integrated PDF
-        photon_integral: (Tensor(1)):
-            the integrated photon, not included in PDF
 
         Returns
         -------
         normalization_factor: Tensor(14)
             The normalization factors per flavour.
         """
-        y = op.flatten(pdf_integrated)
+        y = pdf_integrated[0]  # get rid of the batch dimension
         norm_constants = []
 
         if self._msr_enabled:
-            n_ag = [(1.0 - y[GLUON_IDX[0][0] - 1] - photon_integral[0]) / y[GLUON_IDX[0][0]]] * len(
-                GLUON_IDX
-            )
+            n_ag = [(1.0 - y[IDX['sigma']] - y[IDX['photon']]) / y[IDX['g']]]
             norm_constants += n_ag
 
         if self._vsr_enabled:
-            n_av = [3.0 / y[V_IDX[0][0]]] * len(V_IDX)
-            n_av3 = [1.0 / y[V3_IDX[0][0]]] * len(V3_IDX)
-            n_av8 = [3.0 / y[V8_IDX[0][0]]] * len(V8_IDX)
-            n_av15 = [3.0 / y[V15_IDX[0][0]]] * len(V15_IDX)
+            n_av = [3.0 / y[IDX['v']]] * 3
```

Comment:
Same as before, here I'd rather separate "V" and "V3815" (or something)
so that you can have
```
V = ['v', 'v35', 'v24']
V38  = ['v3', 'v8', 'v15]
```
since the `V` list might actually not be `3` forever.

Reply:
I don't understand what exactly you propose, could you make the change?

Reply:
Here you assume that the same normalization (that of V) is going to be always applied to V, V35 and V24 (you changed the `len(V_IDX)` to just `3`) but this is not necessarily true.

One way around this problem is, since at this stage the content of `pdf_integrated` should be the same for V, V35 and V24, to treat them separately.
i.e., if vsr_enabled, compute all of them separately even if some are the same.

I proposed organizing them in the ones that go with V and the ones that don't, but on second thought is not necessarily better, while separating all of them would also work in the future.

(Please check that this is indeed the case and `pdf_integrated[3] == pdf_integrated[7] == pdf_integrated[8]`)

Reply:
Actually `pdf_integrated[7] == pdf_integrated[8]`, but `pdf_integrated[3]` is different. I was worried for a while but there is a good reason, 3 is divided by x in msr.py while the others are not. I assume that is intentional?

How about what I have now? I factored out all the physics to the top, apart from the msr numerator, which would get messy.

Reply:
>  I assume that is intentional?

Since it was not being used, it was not included in the division by x. If it were to be used it should be also included!

> How about what I have now?

I like this new version a lot! It makes it very clear what is going to happen just by looking at the top of the file.
Btw, the definitions of these normalization constants are here https://arxiv.org/pdf/2109.02653.pdf (eq. 3.8) in case you are interested on what they are / why they are there / etc